it/controllers/pensions/paymentsIntoOverseasPensions/PaymentIntoPensionSchemeControllerISpec.scala

```diff
@@ -0,0 +1,501 @@
+          inputFieldValueCheck(amountInputName, amountInputSelector, amountOverMaximum)
+          formPostLinkCheck(paymentsIntoPensionSchemeUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+          errorAboveElementCheck(user.specificExpectedResults.get.overseasPensionSchemeErrorAmountOverMaximum)
+          errorSummaryCheck(user.specificExpectedResults.get.overseasPensionSchemeErrorAmountOverMaximum, expectedAmountErrorHref)
+        }
+
+      }
+    }
+
+    "redirect to did your employer pay question page when user selects 'yes' with a valid amount" which {
+
+      lazy val form: Map[String, String] = Map(
+        RadioButtonAmountForm.yesNo -> RadioButtonAmountForm.yes, RadioButtonAmountForm.amount2 -> existingAmount)
+      lazy val result: WSResponse = {
+        dropPensionsDB()
+        authoriseAgentOrIndividual(isAgent = false)
+        val viewModel = aPaymentsIntoOverseasPensionsViewModel.copy(paymentsIntoOverseasPensionsAmount = None,
+          paymentsIntoOverseasPensionsQuestions = None)
+        insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(
+          paymentsIntoOverseasPensions = viewModel)), aUserRequest)
+        urlPost(fullUrl(paymentsIntoPensionSchemeUrl(taxYearEOY)), body = form, follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+
+      }
+
+      "has a SEE_OTHER(303) status" in {
+        result.status shouldBe SEE_OTHER
+        //todo redirect to did your employer pay into your overseas pension scheme
+        result.header("location") shouldBe Some(paymentsIntoPensionSchemeUrl(taxYearEOY))
+      }
+
+      "updates payment into overseas pension page to  Some(true) with valid amount" in {
+        lazy val cyaModel = findCyaData(taxYearEOY, aUserRequest).get
+        cyaModel.pensions.paymentsIntoOverseasPensions.paymentsIntoOverseasPensionsQuestions shouldBe Some(true)
+        cyaModel.pensions.paymentsIntoOverseasPensions.paymentsIntoOverseasPensionsAmount shouldBe Some(BigDecimal(existingAmount))
+      }
+    }
+
+    "redirect to did your employer pay question page when user selects 'No' " which {
+
+      lazy val form: Map[String, String] = Map(
+        RadioButtonAmountForm.yesNo -> RadioButtonAmountForm.no, RadioButtonAmountForm.amount2 -> "")
+      lazy val result: WSResponse = {
+        dropPensionsDB()
+        authoriseAgentOrIndividual(isAgent = false)
+        val viewModel = aPaymentsIntoOverseasPensionsViewModel.copy(paymentsIntoOverseasPensionsAmount = None,
+          paymentsIntoOverseasPensionsQuestions = None)
+        insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(
+          paymentsIntoOverseasPensions = viewModel)), aUserRequest)
+        urlPost(fullUrl(paymentsIntoPensionSchemeUrl(taxYearEOY)), body = form, follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+
+      }
+
```

Comment:
We do not have a test case for user selecting No and an amount is already there. Let's say the page get's populated with Yes and an amount. But user selects No and then hits the Submit button.